server/src/test/java/org/elasticsearch/common/settings/SettingsTests.java

```diff
@@ -441,6 +442,17 @@ public void testWriteSettingsToStream() throws IOException {
         assertEquals(Arrays.asList("1", "2"), settings.getAsList("test.key4.foo"));
     }
 
+    public void testDiff() throws IOException {
```

Comment:
Can we have some more tests? eg a noop test, a test with more than one setting, etc. Basically, can we have something to exercise each path in the setting specific diff code you added (maybe the noop is the only additional thing needed).